game-core/src/main/java/games/strategy/triplea/ai/pro/util/MoveBatcher.java

```diff
@@ -0,0 +1,164 @@
+package games.strategy.triplea.ai.pro.util;
+
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Unit;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * MoveBatcher implements an algorithm that batches AI moves together, reducing the number of
+ * distinct moves the AI does (and thus the visual delay) by combining moves together.
+ */
+public class MoveBatcher {
+  private static class Move {
+    private final ArrayList<Unit> units;
+    private final Route route;
+    private final HashMap<Unit, Unit> unitsToTransports;
+
+    private Move(
+        final ArrayList<Unit> units,
+        final Route route,
+        final HashMap<Unit, Unit> unitsToTransports) {
+      this.units = units;
+      this.route = route;
+      this.unitsToTransports = unitsToTransports;
+    }
+
+    Move(final ArrayList<Unit> units, final Route route) {
+      this(units, route, null);
+    }
+
+    Move(final Unit unit, final Route route, final Unit transportToLoad) {
+      this(
+          mutableSingletonList(unit),
+          route,
+          new HashMap<Unit, Unit>(Map.of(unit, transportToLoad)));
```

Comment:
```suggestion
          new HashMap<>(Map.of(unit, transportToLoad)));
```
Pretty sure that works